milight_ibox2/milight_ibox2_control.py

```diff
@@ -109,7 +116,49 @@ def print_bytearray(data, msg=""):
     # ----------------------------------------------------------------------------------------------
     # Milight iBox2 functions
     # ----------------------------------------------------------------------------------------------
-    def connect(self):
+    def scan(self):
+        """ Scan and return all iBox2 devices in network
+        :return: List: [{'ip': '10.10.100.254', 'port': 5987, 'mac': 'F0:FE:6B:XX:XX:XX'}, ...]
+        """
+        found_ibox_devices = []
+        data = bytearray([0x13, 0x00, 0x00, 0x00, 0x0a, 0x03, 0x9b, 0x7f, 0x11, 0xf0, 0xfe, 0x6b, 0x3b, 0xdd, 0xd4])
+
+        self.socket_open()
+        self.socket_send(data, broadcast=True)
+
+        while 1:
+            try:
+                # Wait for UDP response from iBox
+                (data, (ibox_addr, ibox_port)) = self.sock_server.recvfrom(100)
```

Comment:
potential runtime error if none is received

Reply:
I agree, and I've pushed a fix to this branch. 

Note: I could not find a situation to return type `None`. In Python3.9 an empty `bytearray` is returned.